code/graphics/render.cpp

```diff
@@ -1193,33 +1193,71 @@ static void draw_bitmap_list(bitmap_rect_list* list, int n_bm, int resize_mode,
 	auto vert_list = new vertex[6 * n_bm];
 	float sw = 0.1f;
 
+
+	float rad = 0;
+
+	if (angle != 0) {
+		rad = fl_radians(angle);
+	}
+
 	for (int i = 0; i < n_bm; i++) {
 		// stuff coords
-
+		
 		bitmap_2d_list* b = &list[i].screen_rect;
 		texture_rect_list* t = &list[i].texture_rect;
+		
+		float centerX = 0;
+		float centerY = 0;
+
+		if (angle != 0.f) {
+			centerX = (b->x + (b->x + b->w)) / 2.0f;
+			centerY = (b->y + (b->y + b->h)) / 2.0f;
+		}
+
 		// tri one
 		vertex* V = &vert_list[i * 6];
-		V->screen.xyw.x = (float)b->x;
-		V->screen.xyw.y = (float)b->y;
+		
+		if (angle != 0.f) {
+			V->screen.xyw.x = cosf(rad) * (b->x - centerX) - sinf(rad) * (b->y - centerY) + centerX;
+			V->screen.xyw.y = sinf(rad) * (b->x - centerX) + cosf(rad) * (b->y - centerY) + centerY;
```

Comment:
I just want to double-check that the center point is calculated correctly.  Is the x,y point always going to be the top-left corner of the image, even when the image is rotated?  If you draw the same image with a 0 angle and with a rotated angle, will their top-left corners meet at the same point?

Or is it instead the case that rotated images are drawn with the x,y representing their center point?  In fact, thinking along those lines, it would be handy to have an alternate function that uses the center point and the width/height, in addition to the x1,y1-x2,y2 method.

Reply:
Here's [a screenshot](https://i.imgur.com/TskqmEK.png) of how it looks when a rotated and a non-rotated image are in the same place. The one in the foreground is the non-rotating one. 

Without this calculation, the image would rotate around the top left, which would effectively change its position while it rotates. 

I agree that it'd be nice to have a version that uses width/height, because IMO the x2/y2 is a bit unintuitive.

Reply:
I can't really tell how those images are aligned - if they share a corner, or share a center, or share width/height.

I agree that the image shouldn't rotate around its top-left corner, but I'm not sure what it is actually doing.

Reply:
They share a center